### DISCUSSION TITLE
Enabling peripheral support for esp32 s2

### DISCUSSION THREAD

**Discussion by AnandKenshin** [zephyrproject-rtos/zephyr#82636] [Q&A]
Hi All,

I am new to zephyrRTOS and project requires below listed peripherals.
CAN
WIFI
socket
HTTP server and client
Would like to know if those are already enabled on any .dts files.
Or else would like to know about enabling these peripherals via prj.conf or by using a .overlay file

I noticed these on esp32s2_common.dtsi file
wifi: wifi {
		compatible = "espressif,esp32-wifi";
		status = "disabled";
	};

		twai: can@3f42b000 {
			compatible = "espressif,esp32-twai";
			reg = <0x3f42b000 DT_SIZE_K(4)>;
			interrupts = <TWAI_INTR_SOURCE IRQ_DEFAULT_PRIORITY 0>;
			interrupt-parent = <&intc>;
			clocks = <&rtc ESP32_TWAI_MODULE>;
			status = "disabled";
		};
Not sure if I enable these it would fix the issue		
Would appreciate any help.

Thanks and Regards,
Anand

**Comment by rftafas** [comment]
For supported features in each device: https://github.com/zephyrproject-rtos/zephyr/issues/29394 - List is not fully complete and some parts may need further clarification. Ask there if needed.

If you are new to Zephyr, I recommend "The Pull-up Resistor Channel" videos. They are a little bit outdated after v3.6, but still educational. Here: https://www.youtube.com/@thepullupresistor

You can add that as an overlay to a supported board. Worth reading: https://docs.zephyrproject.org/latest/build/dts/howtos.html

Worth checking: https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/espressif

And: https://docs.zephyrproject.org/latest/boards/espressif/esp32s2_devkitc/doc/index.html

**Comment by AnandKenshin** [comment]
Thanks for the info.
Had a look at the pull up resistor channel before.
Will look into the device tree documentation.
